Algorithms.Test/ShuffleTheArrayTest.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Algorithms.Test
+{
+    public class ShuffleTheArrayTest
+    {
+        private ShuffleTheArray _shuffleTheArray;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public ShuffleTheArrayTest() {
+            _shuffleTheArray    = new ShuffleTheArray();
+        }
+
+        [Fact]
+        public void TestCase1() {
+            //Input
+            int[] numsArray     = new int[] { 2, 5, 1, 3, 4, 7 };
+            int elements        = 3;
+            //Expected Output
+            int[] outputArray   = new int[] { 2, 3, 5, 4, 1, 7 };
+            var result      = _shuffleTheArray.Shuffle(numsArray, elements);
+            Assert.Equal(outputArray, result);
+        }
+        [Fact]
+        public void TestCase2() {
+            //Input
+            int[] numsArray     = new int[] { 1, 2, 3, 4, 4, 3, 2, 1 };
+            int elements        = 4;
+            //Expected Output
+            int[] outputArray   = new int[] { 1, 4, 2, 3, 3, 2, 4, 1 };
+            var result      = _shuffleTheArray.Shuffle(numsArray, elements);
+            Assert.Equal(outputArray, result);
+        }
+        [Fact]
+        public void TestCase3() {
+            //Input
+            int[] numsArray     = new int[] { 1, 1, 2, 2 };
+            int elements        = 2;
+            //Expected Output
+            int[] outputArray   = new int[] { 1, 2, 1, 2 };
+            var result      = _shuffleTheArray.Shuffle(numsArray, elements);
+            Assert.Equal(outputArray, result);
+        }
+        [Fact]
+        public void TestCase4() {
```

Comment:
Imagine this is a real project. TestCase4 doesn't tell me what this test actually do. You can check unit tests best practices here:
https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests

Reply:
I check the information 💯